bundles/org.eclipse.jface/src/org/eclipse/jface/dialogs/DialogSettings.java

```diff
@@ -87,6 +88,8 @@ public class DialogSettings implements IDialogSettings {
 
 	// A Map with all the keys mapped to array of strings.
 	private Map<String, String[]> arrayItems;
+	private LinkedList<String> lastFiveSearches;
```

Comment:
Using a LinkedList is of discouraged. See the example below.
In this case using a `Deque` is maybe more suitable?
```suggestion
	private Deque<String> lastFiveSearches = new ArrayDequeue();
```

Reply:
done